homeassistant/components/rainmachine/sensor.py

```diff
@@ -162,3 +195,22 @@ def update_from_latest_data(self) -> None:
         """Update the state."""
         if self.entity_description.key == TYPE_FREEZE_TEMP:
             self._attr_native_value = self.coordinator.data["freezeProtectTemp"]
+
+
+class ZoneTimeRemainingSensor(RainMachineEntity, SensorEntity):
+    """Define a sensor that shows the amount of time remaining for a zone."""
+
+    entity_description: RainMachineSensorDescriptionUid
+
+    @callback
+    def update_from_latest_data(self) -> None:
+        """Update the state."""
+        now = utcnow()
+        seconds_remaining = self.coordinator.data[self.entity_description.uid][
+            "remaining"
+        ]
+
+        if seconds_remaining == 0:
+            self._attr_native_value = None
```

Comment:
Is this the appropriate thing to do when there are `0` seconds remaining (i.e., the default for zones that aren't currently running)?

Reply:
Is it valuable to know when the last run completed? If so we should leave it to be when it finished

Reply:
Maybe name it zone completion time instead?